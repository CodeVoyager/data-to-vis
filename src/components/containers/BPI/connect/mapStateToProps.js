import {
    availableCurrencies,
    currency,
    data,
    startDate,
    endDate,
    highlights,
    currentHighlight
} from '../../../../selectors/bpi';
import {
    isLoading,
} from '../../../../selectors';

const mapStateToProps = state => {
    return {
        availableCurrencies: availableCurrencies(state),
        currency: currency(state),
        data: data(state),
        startDate: startDate(state),
        endDate: endDate(state),
        isLoading: isLoading(state),
        highlights: highlights(state),
        currentHighlight: currentHighlight(state),
    };
}

export default mapStateToProps;