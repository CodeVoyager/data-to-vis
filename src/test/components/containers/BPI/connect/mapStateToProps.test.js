import mapStateToProps from '../../../../../components/containers/BPI/connect/mapStateToProps';
import moment from 'moment';


describe('BPI mapStateToProps', () => {

    it('should produce valid results for empty state', () => {
        const state = {};

        expect(mapStateToProps())
            .toEqual({
                availableCurrencies: [],
                currency: null,
                data: [],
                startDate: null,
                endDate: null,
                isLoading: false,
                highlights: [],
                currentHighlight: null,
            });
        expect(mapStateToProps(state))
            .toEqual({
                availableCurrencies: [],
                currency: null,
                data: [],
                startDate: null,
                endDate: null,
                isLoading: false,
                highlights: [],
                currentHighlight: null,
            });
    });

    it('should produce valid results for valid state', () => {
        const state1 = {
            loadingCounter: 0,
            bpi: {
                availableCurrencies: ['USD'],
                currency: 'USD',
                data: [],
                startDate: moment('20171021', 'YYYYMMDD'),
                endDate: moment('20171023', 'YYYYMMDD'),
                highlights: [],
                currentHighlight: null,
            }
        };
        const state2 = {
            loadingCounter: 4,
            bpi: {
                availableCurrencies: ['USD'],
                currency: 'USD',
                data: [],
                startDate: moment('20171021', 'YYYYMMDD'),
                endDate: moment('20171023', 'YYYYMMDD'),
                highlights: [],
                currentHighlight: {
                    date: moment('20171022', 'YYYYMMDD'),
                    description: 'DESCRIPTION'
                },
            }
        };

        expect(mapStateToProps(state1))
            .toEqual({
                availableCurrencies: ['USD'],
                currency: 'USD',
                data: [],
                startDate: moment('20171021', 'YYYYMMDD'),
                endDate: moment('20171023', 'YYYYMMDD'),
                isLoading: false,
                highlights: [],
                currentHighlight: null,
            });

        expect(mapStateToProps(state2))
            .toEqual({
                availableCurrencies: ['USD'],
                currency: 'USD',
                data: [],
                startDate: moment('20171021', 'YYYYMMDD'),
                endDate: moment('20171023', 'YYYYMMDD'),
                isLoading: true,
                highlights: [],
                currentHighlight: {
                    date: moment('20171022', 'YYYYMMDD'),
                    description: 'DESCRIPTION'
                },
            });

        expect(mapStateToProps(state1)).toMatchSnapshot();
        expect(mapStateToProps(state2)).toMatchSnapshot();
    });

});
