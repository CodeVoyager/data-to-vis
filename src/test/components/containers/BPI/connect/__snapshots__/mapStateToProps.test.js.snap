// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BPI mapStateToProps should produce valid results for valid state 1`] = `
Object {
  "availableCurrencies": Array [
    "USD",
  ],
  "currency": "USD",
  "currentHighlight": null,
  "data": Array [],
  "endDate": "2017-10-22T22:00:00.000Z",
  "highlights": Array [],
  "isLoading": false,
  "startDate": "2017-10-20T22:00:00.000Z",
}
`;

exports[`BPI mapStateToProps should produce valid results for valid state 2`] = `
Object {
  "availableCurrencies": Array [
    "USD",
  ],
  "currency": "USD",
  "currentHighlight": Object {
    "date": "2017-10-21T22:00:00.000Z",
    "description": "DESCRIPTION",
  },
  "data": Array [],
  "endDate": "2017-10-22T22:00:00.000Z",
  "highlights": Array [],
  "isLoading": true,
  "startDate": "2017-10-20T22:00:00.000Z",
}
`;
